src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,228 @@
+            }
+            node = node.next;
+        }
+
+
         return null;
     }
 
+
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        int newCapacity = CAPACITY * 2;
+        Node<K, V>[] newTable = new Node[newCapacity];
+        int index = 0;
+        int sizeOfArr = CAPACITY;
+        CAPACITY = newCapacity;
+
+        for (int i = 0; i < sizeOfArr; i++) {
+            Node<K, V> node = table[i];
+            //linked list can be empty
+            //if node == null linked list is empty
+            if (node != null) {
+                index = hash(node.key);
+                newTable[index] = node;
+            } else {
+                //if node == null linked list is empty skip this
+                continue;
+            }
+
+            //old linked list isn't empty and has more than 1 elem-> we should go by all node until next == null
+            if (node.next != null) {
+                //if linked list has one element if it fails???
+                while (node.next != null) {
+                    Node<K, V> nextNode = node.next;
+                    //hash doesn't work correctly (the elements have the )
+                    index = hash(nextNode.key);
+                    //new linked list is empty
+                    if (newTable[index] == null) {
+                        newTable[index] = nextNode;
+                        //what should we do if not
+                    } else {
+                        //does it work correctly?
+                        //seems yes
+                        //newTable[index].next = nextNode;
+                    }
+                    node = node.next;
+                }
+            }
+
+        }
+
+        table = newTable;
+        threshold = (int) (CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
+
+//    private void putPair(K key, V value) {
+//        int index = hash(key);
+//        Node<K, V> newNode = new Node<>(key, value, null);
+//        Node<K, V> node = table[index];
+//        //linked list is empty
+//        if (node == null) {
+//            table[index] = newNode;
+//            size++;
+//            //linked list isn't empty
+//        } else {
+//            //оно не заходит сюда
+//            //node.key.equals(key)
+//            while (node.next != null || checkKey(key, node)) {
+//                if (checkKey(key, node)) {
+//                    node.value = value;
+//                    return;
+//                }
+//                node = node.next;
+//            }
+//            //okay if we don't have the same keys -> we should put new pair
+//            node.next = newNode;
+//            size++;
+//        }
+//    }
+
+    private boolean checkKey(K key, Node<K, V> node) {
+        if (node.key == key || node.key != null && node.key.equals(key)) {
+            return true;
+        }
+        return false;
+    }
+
+
+    static class Node<K, V> implements Map.Entry<K, V> {
```

Comment:
According to the checklist, do not use getters and setters in class Node. You have access to private fields of inner class, so using getters and setters is redundant.